src/filebuf.c

```diff
@@ -71,6 +71,21 @@ static int lock_file(git_filebuf *file, int flags, mode_t mode)
 		char buffer[FILEIO_BUFSIZE];
 		ssize_t read_bytes;
 
+#ifdef GIT_WIN32
+		bool hidden = false;
+		if (strstr(file->path_original, ".git") == NULL) {
```

Comment:
Indeed it will, do you think we could change this to:

strstr(file->path_original, "/.git")

to solve that issue?